elevation_mapping/src/ElevationMapping.cpp

```diff
@@ -237,6 +238,20 @@ void ElevationMapping::visibilityCleanupThread()
 void ElevationMapping::pointCloudCallback(
     const sensor_msgs::PointCloud2& rawPointCloud)
 {
+  // Check if point cloud has corresponding robot pose at the beginning
+  if(not initialized_) {
+    double oldest_pose_time = robotPoseCache_.getOldestTime().toSec();
+    double current_pcl_time = rawPointCloud.header.stamp.toSec();
+
+    if(current_pcl_time < oldest_pose_time) {
+      ROS_WARN("No corresponding point cloud and pose are found. Waiting for first match.");
+      return;
+    } else {
+      ROS_WARN("First corresponding point cloud and pose found, initialized. ");
```

Comment:
You could use `ROS_DEBUG_THROTTLE` to limit amount of messages shown.

Reply:
This should only print once, as receivedFirstMatchingPointcloudAndPose_ is set to true afterwards.

Reply:
I didn't know about this. Dankeschön Harmish.

Regarding the value that sets the timout for the warn throttle (5 seconds at the moment). I think it could be fetched from the ROS parameter server, but that may be a bit overkill.